### PR TITLE
Moving forward to the latest HF transformers

### PR DESCRIPTION
**Pull request by drnikolaev**
ONNX export verified.

### CODE REVIEW

File: collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py [hunk 1]
```diff
@@ -263,7 +263,7 @@ def truncate_seq_pair(a, b, max_num_tokens):
 
         # TODO: wrap the return value with () for consistent style.
         return np.array(input_ids), input_type_ids,\
-            np.array(input_mask, dtype=np.float32), np.array(output_ids),\
+            np.array(input_mask, dtype=np.long), np.array(output_ids),\
```
Comment (chiphuyen) [inline collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py]: will have to update this on all other datasets in nemo_nlp that use bert too

File: examples/nlp/scripts/get_wkt2.sh [hunk 2]
```diff
@@ -7,14 +7,15 @@ Copyright by the AWD LSTM authors.
 
 echo "- Downloading WikiText-2"
 
+cd /raid
 wget --continue -P data/lm/ https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip
 unzip -q data/lm/wikitext-2-v1.zip -d data/lm
-cd data/lm/wikitext-2
+cd /raid/data/lm/wikitext-2
```
Comment (chiphuyen) [inline examples/nlp/scripts/get_wkt2.sh]: this only works if users have /raid on their machine

Comment (drnikolaev) [reply examples/nlp/scripts/get_wkt2.sh]: Fixed, thank you

File: collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py (hunk 1 above)
Comment (drnikolaev) [reply collections/nemo_nlp/nemo_nlp/data/datasets/bert_pretraining.py]: done

**Comment by okuchaiev** [comment]
"Also it breaks Bert pretraining with O1, the loss is not doing down as expected but is stuck at higher than expected (~5 instead of ~2 after same number of iterations"
- This sounds like a big problem. 
- Do we have this problem for O2 as well?
-Let's count the number of trainable parameters here and in the previous version to make sure the models are indeed exactly the same.

**Comment by yzhang123** [comment]
> "Also it breaks Bert pretraining with O1, the loss is not doing down as expected but is stuck at higher than expected (~5 instead of ~2 after same number of iterations"
> 
> * This sounds like a big problem.
> * Do we have this problem for O0 as well?
>   -Let's count the number of trainable parameters here and in the previous version to make sure the models are indeed exactly the same.

i tried O0, but loss is also stuck at ~5. i checked both model parameters, and they are both identical. So not sure what the problem is.

**Comment by drnikolaev** [comment]
On my box:
```
2019-12-04 12:58:49,980 - INFO - Step: 3675
2019-12-04 12:58:49,980 - INFO - Loss: 7.059 MLM Loss: 6.284 NSP Loss: 0.775
2019-12-04 12:58:49,980 - INFO - Step time: 0.5305397510528564 seconds
2019-12-04 12:59:03,323 - INFO - Step: 3700
2019-12-04 12:59:03,323 - INFO - Loss: 6.955 MLM Loss: 6.263 NSP Loss: 0.692
2019-12-04 12:59:03,323 - INFO - Step time: 0.5290660858154297 seconds
2019-12-04 12:59:13,281 - INFO - Finished epoch 9 in 200.57236099243164
2019-12-04 12:59:13,282 - INFO - Finished epoch 9 in 200.57336378097534
2019-12-04 12:59:13,901 - INFO - Saved checkpoint: ./tmp/2019-12-04_12-25-28/checkpoints/trainer-EPOCH-9.pt
2019-12-04 12:59:13,901 - INFO - Done in 2019.4554471969604
2019-12-04 12:59:14,508 - INFO - Saved checkpoint: ./tmp/2019-12-04_12-25-28/checkpoints/trainer-EPOCH-10.pt
```

**Comment by ekmb** [comment]
@drnikolaev could you please merge with the latest master and convert the input_mask type in this new example as well https://github.com/NVIDIA/NeMo/blob/master/collections/nemo_nlp/nemo_nlp/data/datasets/punctuation_capitalization.py#L389

**Pull request merged**